Add docs for saturating integer arithmetic.

jbcrail opened the pull request:
This was added for #23241.

rust-highfive commented:
r? @huonw

(rust_highfive has picked a reviewer for you, use r? to override)

steveklabnik commented on src/libcore/num/mod.rs:
```diff
@@ -345,6 +345,16 @@ pub trait Int
     /// Saturating integer addition. Computes `self + other`, saturating at
     /// the numeric bounds instead of overflowing.
+    ///
+    /// # Example
```
These should be 'examples'

steveklabnik commented on src/libcore/num/mod.rs:
````diff
@@ -345,6 +345,16 @@ pub trait Int
+    ///
+    /// # Example
+    ///
+    /// ```rust
````
you don't need the `rust` for API documentation

steveklabnik commented:
r=me with two nits, thanks!

jbcrail commented:
Thanks for the comments. I've updated the branch.

On a related note, I noticed other examples have the same issues that you mentioned. Would it be too pedantic if I corrected those in another pull request?

steveklabnik commented:
I've been trying to play 'boy scout rule' with them, but if you'd like to clean them up, I'd love that.

steveklabnik commented:
oh, and would you mind squashing these please? thanks

jbcrail commented:
I squashed the commits. I also sent a pull request to remove the Rust syntax highlighting from the example docs.

jbcrail commented:
Is anything else needed to close this pull request? Thanks.

steveklabnik commented:
@bors: r+

bors commented:
:pushpin: Commit 62be565 has been approved by `steveklabnik`

<!-- @bors r=steveklabnik 62be565811e9805ec1045395eef2f977d7385efb -->

steveklabnik commented:
Whoops, sorry about that! Sometimes these things slip. Thank you!

bors commented:
:hourglass: Testing commit 62be565 with merge e98e391...

bors commented:
:sunny: Test successful - [auto-linux-32-nopt-t](http://buildbot.rust-lang.org/builders/auto-linux-32-nopt-t/builds/4130), [auto-linux-32-opt](http://buildbot.rust-lang.org/builders/auto-linux-32-opt/builds/4134), [auto-linux-64-nopt-t](http://buildbot.rust-lang.org/builders/auto-linux-64-nopt-t/builds/4116), [auto-linux-64-opt](http://buildbot.rust-lang.org/builders/auto-linux-64-opt/builds/4124), [auto-linux-64-x-android-t](http://buildbot.rust-lang.org/builders/auto-linux-64-x-android-t/builds/4097), [auto-mac-32-opt](http://buildbot.rust-lang.org/builders/auto-mac-32-opt/builds/4152), [auto-mac-64-nopt-t](http://buildbot.rust-lang.org/builders/auto-mac-64-nopt-t/builds/4150), [auto-mac-64-opt](http://buildbot.rust-lang.org/builders/auto-mac-64-opt/builds/4137), [auto-win-32-nopt-t](http://buildbot.rust-lang.org/builders/auto-win-32-nopt-t/builds/3810), [auto-win-32-opt](http://buildbot.rust-lang.org/builders/auto-win-32-opt/builds/3797), [auto-win-64-nopt-t](http://buildbot.rust-lang.org/builders/auto-win-64-nopt-t/builds/2627), [auto-win-64-opt](http://buildbot.rust-lang.org/builders/auto-win-64-opt/builds/2634)
